From 33d8a5aa92e3921e0c66b3db9a2542425b5b3e96 Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Fri, 19 Sep 2025 13:19:24 +0100
Subject: [PATCH] EGLUtils: Ensure we have at least 8 alpha bits

Previously R10G10B10A2 will be preferred over R8G8B8A8 is supported by MESA
But 2 bits of alpha looks very bad in kodi when transparency effects are used.

Ensure we have at leats 8 bits of alpha.

Fixes: b902f8bd: EGLUtils: rework the way we choose the egl config
---
 xbmc/utils/EGLUtils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/utils/EGLUtils.cpp b/xbmc/utils/EGLUtils.cpp
index 1d7ba97741254..1832271663652 100644
--- a/xbmc/utils/EGLUtils.cpp
+++ b/xbmc/utils/EGLUtils.cpp
@@ -286,7 +286,7 @@ bool CEGLContextUtils::ChooseConfig(EGLint renderableType, EGLint visualId, bool
   attribs.Add({{EGL_RED_SIZE, 8},
                {EGL_GREEN_SIZE, 8},
                {EGL_BLUE_SIZE, 8},
-               {EGL_ALPHA_SIZE, 2},
+               {EGL_ALPHA_SIZE, 8},
                {EGL_DEPTH_SIZE, 16},
                {EGL_STENCIL_SIZE, 0},
                {EGL_SAMPLE_BUFFERS, 0},
